hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -401,6 +406,49 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 			log.Info("hostedcluster is still deleting", "name", req.NamespacedName)
 			return ctrl.Result{RequeueAfter: clusterDeletionRequeueDuration}, nil
 		}
+
+		// This new condition is necessary for OCM personnel to report any cloud dangling objects to the user.
+		// The grace period is customizable using the flag --destroy-grace-period. It's a time.Duration field type.
+		// This condition only will be created when !SelfManage and CloudResourcesDestroyedCondition is False.
+		CloudResourcesDestroyedCondition := meta.FindStatusCondition(hcluster.Status.Conditions, string(hyperv1.CloudResourcesDestroyed))
+		if CloudResourcesDestroyedCondition.Status == "False" {
```

Comment:
I was thinking that we would always report the HostedClusterDestroyed condition, not just in the case that CloudResourcesDestroyed is false